lib/Entity.js

```diff
@@ -262,6 +298,34 @@ export class Entity {
         }
     }
 
+    /**
+     * Internal update method intended to be overridden by derived class.
+     * Separate logic from rendering by updating the derived class using this method.
+     */
+    internalUpdate() {
+
+    }
+
+    /**
+     * Public-facing update method, not intended to be overridden by derived classes.
+     */
+    update() {
+        if (this.#pen.frameCount === this.#frameLastUpdated) {
+            return;
+        }
+        this.internalUpdate();
+        this.#frameLastUpdated = this.#pen.frameCount;
+
+        if (this.ageless) {
+            return;
+        }
+        if (this.#lifespan > 0) {
```

Comment:
Even though it's not necessary, did we want to make this check if lifespan is not Infinity before decrementing as part of the if statement? That way we have a reminder here that the lifespans default value is Infinity

Reply:
Done.